mantidproject/mantid
# SaveAscii2 - SpectrumMetaData algorithm property

The SaveAscii2 algorithm now has a new property `SpectrumMetaData`. This property enables additional meta data to be output to the ascii file. Currently the only valid inputs for the additional meta data are SpectrumNumber, Q and Angle.

**To test:**

Code review:
- The way I've implemented this might not be the best way. It's currently a map that is used to store the relevant SpecNo,Q and angle values which are then printed in the ascii file. If you can think of a better way of doing this, let me know.
- The `WriteSpectrumID` property is still in this algorithm. This can be duplicated by having `SpectrumNumber` in the SpectrumMetaData property and hence makes the `WriteSpectrumID` property somewhat redundant. It'd be good to hear thoughts on this as well.

Functional testing:
- Load in a workspace that has units of Q 
  - `irs26176_graphite002_red` is a possible data set
- Execute SaveAscii with the SpectrumMetaData of SpectrumNumber,Q,Angle (case should not matter)
- Show the detectors on the workspace you input to save ascii.
- Load the saved ascii file
- Ensure Angle(ascii) == Theta(det_table)
- Ensure `Q == Q`

Fixes #15623 

[Release Notes](https://github.com/mantidproject/mantid/blob/15623_SaveAscii_spectrumMetaData/docs/source/release/v3.7.0/framework.rst#id71)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
